src/quo2/components/channel_avatar.cljs

```diff
@@ -0,0 +1,47 @@
+(ns quo2.components.channel-avatar
+  (:require [quo.design-system.colors :as colors]
+            [quo2.foundations.colors :as f-colors]
+            [quo.react-native :as rn]
+            [clojure.string :as clj-string]
+            [status-im.ui.components.icons.icons :as icons]))
+
+(defn channel-avatar [{:keys [big? dark? lock-status icon x y]}]
+  [rn/view {:style {:width (if big? 32 24)
+                    :height (if big? 32 24)
+                    :top (if (or (clj-string/blank? x)
+                                 (js/isNaN x))
+                           0
+                           (js/parseInt x))
+                    :left (if (or (clj-string/blank? y)
+                                  (js/isNaN y))
+                            0
+                            (js/parseInt y)) 
+                    :border-radius 100
+                    :background-color (if dark? "#131E37"
+                                          "#EDF0FC")}}
+   [rn/view {:style {:left (if big? 6 3)
+                     :top (if big? 6 3)
+                     :width 20
+                     :height 20}}
+    [rn/view {:style {:top 2.5
```

Comment:
Use [Flexbox](https://reactnative.dev/docs/flexbox) layout, to avoid using extra nested views.

Reply:
Done